tests/Doctrine/Tests/ORM/Functional/Ticket/Issue6313Test.php

```diff
@@ -0,0 +1,245 @@
+<?php
+namespace Doctrine\Tests\ORM\Functional\Ticket;
+
+use Doctrine\DBAL\Platforms\AbstractPlatform;
+use Doctrine\DBAL\Types\StringType;
+use Doctrine\DBAL\Types\Type;
+use Doctrine\ORM\AbstractQuery;
+use Doctrine\Tests\OrmFunctionalTestCase;
+use Doctrine\Common\Collections\ArrayCollection;
+
+final class Issue6313Test extends OrmFunctionalTestCase
+{
+    protected function setUp()
+    {
+        parent::setUp();
+
+        $this->_schemaTool->createSchema(
+            [
+                $this->_em->getClassMetadata(Issue6313_Unit::class),
```

Comment:
Please reduce the test case to the minimum possible amount of fields/entities needed to replicate the issue without your patch applied, then please rename entities so they make sense in the context of this test, rather than their origin context (your codebase)